src/main/java/org/folio/auth/authtokenmodule/tokens/AccessToken.java

```diff
@@ -49,7 +49,7 @@ protected Future<Token> validateContext(TokenValidationContext context) {
     try {
       validateCommon(context.getHttpServerRequest());
     } catch (TokenValidationException e) {
-      return Future.failedFuture(e);
+      return handleCrossTenantRequest(e, context);
```

Comment:
please refactor to avoid using exception for handling business cases, I propose to refactor validateCommon(), it should return Future object, it will allow to invoke user service if tenants not matches in validateCommon() and simplify validateContext method

Reply:
@SerhiiNosko please review again. thank you